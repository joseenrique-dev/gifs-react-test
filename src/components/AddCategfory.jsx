import { useState } from 'react';

const AddCategfory = () => {
    const [inputValue, setInputValue] = useState('');
    const onSubmit = (e) => {
        e.preventDefault();
        console.log(inputValue);
    }
  return (
    <div>
        <form onSubmit={(e) => onSubmit(e)}>
            <input 
                type="text" 
                placeholder="Search for a gift"
                value={inputValue}
                onChange={(e) => setInputValue(e.target.value)}
            />
        </form>
    </div>
  )
}

export default AddCategfory
