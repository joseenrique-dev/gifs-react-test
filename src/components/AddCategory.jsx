import { useState } from 'react';

const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');
    
    const onInputChange = (e) =>{
        setInputValue(e.target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length <= 1) return; 
        onNewCategory(inputValue.trim());        
        setInputValue('');           
    }
  return (
    <div>
        <form onSubmit={(e) => onSubmit(e)}>
            <input 
                type="text" 
                placeholder="Search for a gift"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    </div>
  )
}

export default AddCategory
