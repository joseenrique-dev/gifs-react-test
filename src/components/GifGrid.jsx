import { GifItem } from './GifItem';
import {useFetchGifs} from '../hooks/useFetchGifs'; 
import PropTypes  from 'prop-types';

export const GifGrid = ({ category }) => {
    const {
        images,
        loading,
    } = useFetchGifs(category);

    return (
    <>
        <h3>{category}</h3>
        {loading && <h2>Loading...</h2>}
        <div className='card-grid'>
            {
                images.map( img => 
                    (<GifItem
                        key={img.id} 
                        { ...img }
                    />)
                    )
            }
        </div>
    </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
