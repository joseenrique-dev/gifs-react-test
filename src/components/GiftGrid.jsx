import React from 'react'

const GiftGrid = ({ category }) => {

  return (
    <>
        <h3>{category}</h3>   
    </>
  )
}

export default GiftGrid
