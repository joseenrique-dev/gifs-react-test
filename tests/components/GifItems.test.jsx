import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('GifItems tests', () => {
    const fakeTitle = 'Fake title';
    const fakeUrl = 'http://fakeurl.com/';
    it('should render correctly', () => {
        const {container} = render(<GifItem title={ fakeTitle } url={ fakeUrl } />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render the image with the correct url and the indicated ALT', () => {
        render(<GifItem title={ fakeTitle } url={ fakeUrl } />);
        const { alt, src } = screen.getByRole('img');
        expect(src).toBe(fakeUrl);
        expect(alt).toBe(fakeTitle);
        // expect(screen.getByRole('img').src).toBe(fakeUrl);
        // expect(screen.getByRole('img').alt).toBe(fakeTitle);
    });

    it('should render the title', () => {
        render(<GifItem title={ fakeTitle } url={ fakeUrl } />);
        expect(screen.getByText(fakeTitle)).toBeTruthy();
    });
});