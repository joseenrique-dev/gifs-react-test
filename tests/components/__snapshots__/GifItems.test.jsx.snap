// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItems tests should render correctly 1`] = `
<div
  class="card"
>
  <img
    alt="Fake title"
    src="http://fakeurl.com/"
  />
  <p>
    Fake title
  </p>
</div>
`;
